staging/src/k8s.io/client-go/util/workqueue/queue_test.go

```diff
@@ -159,3 +159,72 @@ func TestReinsert(t *testing.T) {
 		t.Errorf("Expected queue to be empty. Has %v items", a)
 	}
 }
+
+func TestFinishProcessingBeforeShutdown(t *testing.T) {
+	q := workqueue.New()
+	q.Add("foo")
+	q.Add("bar")
+
+	firstItem, _ := q.Get()
+	secondItem, _ := q.Get()
+
+	finishedShutdownTest := finishedShutdownTest{}
```

Comment:
why not use a sync.WaitGroup?

Reply:
The goal I was trying to achieve by assigning `finishedShutdownTest.finished` was to verify that `ShutDown` never returned during the test execution until the very end when we've marked all items as `Done`. It's true that using a `WaitGroup` achieves the same goal more robustly, since if we `Wait` before the last `Done` is called, we would fail the test. Conversely, if we're missing a `Done` for any item on the queue: we also fail the test and time out.